src/models/document/base-document-content.ts

```diff
@@ -1077,5 +1064,148 @@ export const BaseDocumentContentModel = types
           // ignore shared models with errors
         }
       });
+    },
+    duplicateTiles(tiles: IDragTileItem[]) {
+      // Determine the row to add the duplicated tiles into
+      const rowIndex = self.getRowAfterTiles(tiles);
+
+      // Find shared models used by tiles being duplicated
+      const sharedModelEntries = Object.values(self.getSharedModelsUsedByTiles(tiles.map(tile => tile.tileId)));
+
+      // Update shared models with new ids
+      interface UpdatedSharedDataSet {
+        sharedModelId: string;
+        attributeIdMap: Record<string, string>;
+        caseIdMap: Record<string, string>;
+      }
+      const updatedSharedModelMap: Record<string, UpdatedSharedDataSet> = {};
+      const newSharedModelEntries: any[] = [];
+      sharedModelEntries.forEach(sharedModelEntry => {
+        // For now, only duplicate shared data sets
+        if (sharedModelEntry.sharedModel.type === "SharedDataSet") {
+          // Determine new ids
+          const sharedDataSet = getSnapshot(sharedModelEntry.sharedModel) as SharedDataSetSnapshotType;
+          const sharedModelId = uniqueId();
+          const dataSetId = uniqueId();
+          const attributeIdMap: Record<string, string> = {};
+          const caseIdMap: Record<string, string> = {};
+          sharedDataSet.dataSet.attributes?.forEach((attr: any) => {
```

Comment:
I would think you could rely on type inference here:
```suggestion
          sharedDataSet.dataSet.attributes?.forEach(attr => {
```
and then you wouldn't need the `any` here and in several similar loops below this.